### ISSUE TITLE
Request: Graceful Shutdown Handling

### ISSUE THREAD

**Issue by Valkyrien04** [JohnVidler/docker-stationeers#2]
Server script, upon the request from the container to restart or stop, should try to save the current server environment before shutting down.

**Comment by JohnVidler** [comment]
Agreed, although how we actually do this without just wrapping the whole binary isn't super clear to me, as the only way to communicate properly with the server binary is either through signals (does it respond to a SIGQUIT?) or commands via stdin.. and I think the server tries to be 'clever' and doesn't enable the interactive commands bit of itself when it detects its not in an interactive terminal session, so it might be a pain.

Any ideas?

**Comment by JohnVidler** [comment]
Note to self, docker emits SIGTERM then SIGKILL

**Comment by Valkyrien04** [comment]
No ideas how, just that it SHOULD be possible if we can pass commands into the server process.

**Comment by JohnVidler** [comment]
Yeah, I have a prototype wrapper using python that traps the signals from docker and passes them through, but it looks like the server binary is currently ignoring terminal input, so we can't do anything clever like sending a preemptive save then shutdown command.

Although that might be possible if we write our own mod, but that's waaaay down the rabbit hole here.

Note that if anyone does want to do that mod, can it please use the RCON standard protocol please 😸
